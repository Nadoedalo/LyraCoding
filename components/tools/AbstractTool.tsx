import { Component } from 'react';
import { toolStore } from "../../store";
import { observer } from "mobx-react";
@observer export class AbstractTool extends Component<any, any> {
    name = ''; // an ID of the tool, camelCase
    title = ''; // title to display to the user
    constructor(props) {
        super(props);
        return this;
    }

    activateTool() {
        toolStore.toggleTool(this.name);
    }
    get classNames() {
        const classNames = [];
        if(toolStore.activeTools[this.name]) {
            classNames.push('active');
        }
        return classNames.join(' ');
    }
    render() {
        return <span onClick={() => this.activateTool()} title={this.title} className={this.classNames}>
            { this.iconElement() }
        </span>;
    }
    iconElement() { // React Icon element
        return <></>;
    }
}