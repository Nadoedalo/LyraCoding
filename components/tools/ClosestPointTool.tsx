import { Component } from 'react';

export class ClosestPointTool extends Component<any, any> {
    render() {
        return <>
            <span>Closest</span>
        </>;
    }
}