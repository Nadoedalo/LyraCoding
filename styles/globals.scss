html,
body {
  padding: 0;
  margin: 0;
  font-family: Roboto, Verdana, sans-serif;
  width: 100%;
  height: 100%;

  > div {
    width: 100%;
    height: 100%;
  }
}

main {
  display: flex;
  flex-direction: row;
  width: 100%;
  height: 100%;

  .toolbar {
    background: #03DAC6;
    width: 3em;
    padding: 0.5em;
    display: flex;
    flex-direction: column;
    justify-content: center;
    text-align: center;
    box-sizing: border-box;
    border-right: 2px solid #3700B3;
    > div {
      display: flex;
      flex-direction: column;
      padding-top: 3em;
      font-size: 2em;
      > span {
        padding: 0.25rem 0;
        cursor: pointer;
        display: flex;
        justify-content: center;
        transition: 0.3s ease;
        &:hover, &.active {
          color: #3700B3;
        }
      }
      &:first-child {
        padding-top: 0;
      }
    }
  }

  .workspace {
    background-color: #EFEFEF;
    background-image: radial-gradient(black 1px, transparent 0);
    background-size: 20px 20px;
    flex-grow: 1;
  }
}