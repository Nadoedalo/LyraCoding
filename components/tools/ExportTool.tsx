import { AbstractTool } from "./AbstractTool";
import { BiExport } from "react-icons/bi";
import { toolsConfig } from "../../configs/tools.config";
import { shapeStore } from "../../store";

export class ExportTool extends AbstractTool {
    title = 'Export to console';
    name = toolsConfig.EXPORT_TOOL;
    constructor(props) {
        super(props);
        return this;
    }
    iconElement(): JSX.Element {
        return <BiExport />
    }
    activateTool() {
        console.log(shapeStore.export());
    }
}