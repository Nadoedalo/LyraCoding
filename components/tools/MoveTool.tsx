import { AbstractTool } from "./AbstractTool";
import { BiMove } from "react-icons/bi";
import { toolsConfig } from "../../configs/tools.config";

export class MoveTool extends AbstractTool {
    title = 'Move Tool';
    name = toolsConfig.MOVE_TOOL;
    constructor(props) {
        super(props);
        this.activateTool();
        return this;
    }
    iconElement(): JSX.Element {
        return <BiMove />
    }
}