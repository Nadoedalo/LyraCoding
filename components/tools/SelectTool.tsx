import { AbstractTool } from "./AbstractTool";
import { GiArrowCursor } from "react-icons/gi";
import { toolsConfig } from "../../configs/tools.config";

export class SelectTool extends AbstractTool {
    title = 'Select Tool';
    name = toolsConfig.SELECT_TOOL;
    constructor(props) {
        super(props);
        this.activateTool();
        return this;
    }
    iconElement(): JSX.Element {
        return <GiArrowCursor />
    }
}