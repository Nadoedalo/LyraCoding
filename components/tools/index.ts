import { SelectTool } from "./SelectTool";
import { MoveTool } from "./MoveTool";
import { ClosestPointTool } from "./ClosestPointTool";
import { ExportTool } from "./ExportTool";
export const tools = [SelectTool, MoveTool, ClosestPointTool, ExportTool];