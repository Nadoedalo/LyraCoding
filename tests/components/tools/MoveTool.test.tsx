import renderer from 'react-test-renderer';
import { describe, expect, it } from "@jest/globals";
import { MoveTool } from "../../../components/tools/MoveTool";

describe('ToolBar Snapshot', () => {
    const moveTool = renderer.create(<MoveTool />,);

    it("Should match Snapshot", () => {
        let tree = moveTool.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("Changes class when activated", () => {
        renderer.act(() => {
            moveTool.root.instance.activateTool();
        });
        let tree = moveTool.toJSON();
        expect(tree).toMatchSnapshot();
    })

})