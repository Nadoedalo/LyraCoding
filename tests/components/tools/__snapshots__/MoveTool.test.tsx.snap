// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToolBar Snapshot Changes class when activated 1`] = `
<span
  className=""
  onClick={[Function]}
  title="Move Tool"
>
  <svg
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    style={
      {
        "color": undefined,
      }
    }
    viewBox="0 0 24 24"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M18 11h-5V6h3l-4-4-4 4h3v5H6V8l-4 4 4 4v-3h5v5H8l4 4 4-4h-3v-5h5v3l4-4-4-4z"
    />
  </svg>
</span>
`;

exports[`ToolBar Snapshot Should match Snapshot 1`] = `
<span
  className="active"
  onClick={[Function]}
  title="Move Tool"
>
  <svg
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    style={
      {
        "color": undefined,
      }
    }
    viewBox="0 0 24 24"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M18 11h-5V6h3l-4-4-4 4h3v5H6V8l-4 4 4 4v-3h5v5H8l4 4 4-4h-3v-5h5v3l4-4-4-4z"
    />
  </svg>
</span>
`;
